internal/nexodus/keys.go

```diff
@@ -2,68 +2,86 @@ package nexodus
 
 import (
 	"fmt"
-	"os"
-	"strings"
-
 	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
-
-	"go.uber.org/zap"
-)
-
-// default key pair file locations (windows needs work)
-const (
-	workdirPublicKeyFile  = "public.key"
-	workdirPrivateKeyFile = "private.key"
-	linuxPublicKeyFile    = "/etc/wireguard/public.key"
-	linuxPrivateKeyFile   = "/etc/wireguard/private.key"
-	darwinPublicKeyFile   = "/usr/local/etc/wireguard/public.key"
-	darwinPrivateKeyFile  = "/usr/local/etc/wireguard/private.key"
-	windowsPublicKeyFile  = "C:/nexd/public.key"
-	windowsPrivateKeyFile = "C:/nexd/private.key"
-	publicKeyPermissions  = 0644
-	privateKeyPermissions = 0600
+	"os"
+	"runtime"
 )
 
-// generateKeyPair a key pair and write them to disk
-func (ax *Nexodus) generateKeyPair(publicKeyFile, privateKeyFile string) error {
+// handleKeys will look for an existing key pair, if a pair is not found this method
+// will generate a new pair and write them to location on the disk depending on the OS
+func (ax *Nexodus) handleKeys() error {
 
-	privateKey, err := wgtypes.GeneratePrivateKey()
+	err := ax.stateStore.Load()
 	if err != nil {
-		return fmt.Errorf("failed to generate private key: %w", err)
+		return err
 	}
+	state := ax.stateStore.State()
 
-	ax.wireguardPubKey = privateKey.PublicKey().String()
-	ax.wireguardPvtKey = privateKey.String()
+	if state.PublicKey != "" && state.PrivateKey != "" {
+		ax.logger.Infof("Existing key pair found in [ %s ]", ax.stateStore)
+	} else {
+		ax.logger.Infof("No existing public/private key pair found, generating a new pair")
+		wgKey, err := wgtypes.GeneratePrivateKey()
+		if err != nil {
+			return fmt.Errorf("failed to generate private key: %w", err)
+		}
+		state.PublicKey = wgKey.PublicKey().String()
+		state.PrivateKey = wgKey.String()
 
-	// TODO remove this debug statement at some point
-	ax.logger.Debugf("Public Key [ %s ] Private Key [ %s ]", ax.wireguardPubKey, ax.wireguardPvtKey)
-	// write the new keys to disk
-	WriteToFile(ax.logger, ax.wireguardPubKey, publicKeyFile, publicKeyPermissions)
-	WriteToFile(ax.logger, ax.wireguardPvtKey, privateKeyFile, privateKeyPermissions)
+		err = ax.stateStore.Store()
+		if err != nil {
+			return fmt.Errorf("failed store the keys: %w", err)
+		}
+		ax.logger.Debugf("New keys were written to [ %s ]", ax.stateStore)
+	}
 
+	ax.wireguardPubKey = state.PublicKey
+	ax.wireguardPvtKey = state.PrivateKey
 	return nil
+
 }
 
-// readKeyFile reads the contents of a key file
-func readKeyFile(logger *zap.SugaredLogger, keyFile string) string {
-	if !FileExists(keyFile) {
-		return ""
+// loadLegacyKeys should not be needed after everyone has upgraded to the latest nexd.
+func (ax *Nexodus) loadLegacyKeys() (string, string, error) {
```

Comment:
For new code, can you use `nx`?  `ax` is the old variable name used back when it was apex. I've been trying to migrate to `nx` as we go.